File: TesteAndroidv2/app/src/main/java/com/home/project/testeandroidv2/loginScreen/LoginConfigurator.java
package com.home.project.testeandroidv2.loginScreen;

import java.lang.ref.WeakReference;

public enum LoginConfigurator {
    INSTANCE;

    public void configure(LoginActivity loginActivity){

        LoginRouter loginRouter = new LoginRouter();
        loginRouter.activity = new WeakReference<>(loginActivity);


        LoginPresenter loginPresenter = new LoginPresenter();
        loginPresenter.output = new WeakReference<>(loginActivity);

        LoginInteractor loginInteractor = new LoginInteractor();
        loginInteractor.output = loginPresenter;

        if(loginActivity.loginRouter == null){
            loginActivity.loginRouter = loginRouter;
        }

        if(loginActivity.output == null){
            loginActivity.output = loginInteractor;
        }
    }

}

File: BankApp/app/src/main/java/com/schaefer/bankapp/login_screen/LoginConfigurator.kt
package com.schaefer.bankapp.login_screen

import com.schaefer.bankapp.login_screen.activity.LoginActivity
import com.schaefer.bankapp.login_screen.presenter.LoginPresenterImp
import com.schaefer.bankapp.login_screen.router.LoginRouter
import java.lang.ref.WeakReference

enum class LoginConfigurator {
    INSTANCE;

    fun configure(activity: LoginActivity) {
        val router = LoginRouter()
        router.activity = WeakReference<LoginActivity>(activity)

        val presenter = LoginPresenterImp(activity, activity.applicationContext)
        presenter.loginActivityInput = activity

        if (activity.router == null) {
            activity.router = router
        }

        if (activity.presenter == null) {
            activity.presenter = presenter
        }
    }
}